pkg/security/security.go

```diff
@@ -40,6 +43,23 @@ const (
 	Mock = "Mock" // testing only
 )
 
+// TODO: For 1.8, make sure MeshConfig is updated with those settings,
+// they should be dynamic to allow migrations without restart.
+// Both are critical.
+var (
+	// Require 3P TOKEN disables the use of K8S 1P tokens. Note that 1P tokens can be used to request
+	// 3P TOKENS. A 1P token is the token automatically mounted by Kubelet and used for authentication with
+	// the Apiserver.
+	Require3PToken = env.RegisterBoolVar("REQUIRE_3P_TOKEN", false,
```

Comment:
checking my understanding: We normally check `TOKEN_AUDIENCES.contains(token aud)`. If we enable this we check `token aud == nil || TOKEN_AUDIENCES.contains(token aud)`?

Reply:
right now, 
- if policy is '1st party' - we only accept token aud == "", reject tokens with any aud ( I hope - or we have a security bug ).
- if policy is '3p' - we only accept aud = istio-ca - reject aud="".

After this change:
- default is to accept aud== "" || TOKEN_AUDIENCES.contains( token aud ) - as you said
- if REQUIRE_3P_TOKEN= true - it's the same as current "3p" mode. 

The goal is to allow migration - or multicluster where a mix of tokens are in use.